python/paddle/fluid/trainer.py

```diff
@@ -168,8 +172,17 @@ def train(self,
 
         self._train_by_executor(num_epochs, event_handler, reader, feed_order)
 
-    def test(self, reader):
-        pass
+    def test(self, reader, feed_order=None):
+        """
+        Test the model on given test data
+
+        Args:
+            reader: The reader that yields test data.
+            feed_order: Feeding order of reader. None will following the defining
+                order in program
+        """
+
+        return self._test_by_executor(reader, feed_order, self.test_outputs)
```

Comment:
Why put all code into a separate function?

Reply:
Maybe in the future, we need to support `_test_by_parralle_executor`. Here is a reserved interface for switching between multiple executors.